jetty-jmx/src/main/java/org/eclipse/jetty/jmx/MBeanContainer.java

```diff
@@ -122,6 +114,215 @@ public String getDomain()
     }
 
 
+    /**
+     * <p>Creates an ObjectMBean for the given object.</p>
+     * <p>Attempts to create an ObjectMBean for the object by searching the package
+     * and class name space. For example an object of the type:</p>
+     * <pre>
+     * class com.acme.MyClass extends com.acme.util.BaseClass implements com.acme.Iface
+     * </pre>
+     * <p>then this method would look for the following classes:</p>
+     * <ul>
+     * <li>com.acme.jmx.MyClassMBean</li>
+     * <li>com.acme.util.jmx.BaseClassMBean</li>
+     * <li>org.eclipse.jetty.jmx.ObjectMBean</li>
+     * </ul>
+     *
+     * @param o The object
+     * @return A new instance of an MBean for the object or null.
+     */
+    public Object mbeanFor(Object o)
+    {
+        if (o==null)
+            return null;
+
+        Constructor<?> constructor = findMBeanFor(o.getClass());
+
+        if (constructor==null)
+            return null;
+
+        try
+        {
+            Object mbean;
+            if (constructor.getParameterCount()==0)
+            {
+                mbean = constructor.newInstance();
+                ((ModelMBean)mbean).setManagedResource(o, "objectReference");
+            }
+            else
+            {
+                mbean = constructor.newInstance(o);
+            }
+
+            if (mbean instanceof ObjectMBean)
+                ((ObjectMBean)mbean).setMBeanContainer(this);
+
+            if (LOG.isDebugEnabled())
+                LOG.debug("mbeanFor {} is {}", o, mbean);
+
+            return mbean;
+        }
+        catch(Throwable t)
+        {
+            LOG.warn(t);
+        }
+
+        return null;
+    }
+
+    private Constructor<?> findMBeanFor(Class<?> oClass)
+    {
+        if (oClass==null)
+            return null;
+
+        Constructor<?> constructor = _mbeanFor.get(oClass);
+        if (constructor!=null)
+            return constructor;
+
+        try
+        {
+            String pName = oClass.getPackage().getName();
+            String cName = oClass.getName().substring(pName.length() + 1);
+            String mName = pName + ".jmx." + cName + "MBean";
+
+            Class<?> mClass;
+            try
+            {
+                // Look for an MBean class from the same loader that loaded the original class
+                mClass = (Object.class.equals(oClass))?oClass = ObjectMBean.class:Loader.loadClass(oClass, mName);
```

Comment:
You're not wrong... it's the original code but I'm not even sure why it is assigning to oClass?    This will only happen on the last iteration, so oClass should not be reused anyway?     I think the assignment is not needed... and if it is, then it needs to be clearer why.

I'm not even sure we should do any test here... it may be better to just detect Object.class in the top (or bottom) of the loop and return the constructor... or we could precache Object?